relayer/src/jobs/verification.rs

```diff
@@ -1,31 +1,86 @@
-use std::fs;
-
+use crate::domains::{github::get_connection, utils::SBError};
 use anchor_client::{
     anchor_lang::{prelude::Signer, solana_program::example_mocks::solana_sdk::signature::Keypair},
     solana_sdk::{pubkey::Pubkey, signature::read_keypair_file},
 };
-
+use octocrab;
+use serde::{Deserialize, Serialize};
+use serde_json::*;
+use std::{
+    fs::{self, File},
+    result::Result,
+};
 pub struct UserLink {
     domain: String,
     user_profile: String,
     wallet: Pubkey,
 }
 
-pub enum RelayerError {}
-
 /// get_domains calls the rewards contract and get the
 /// potential domains
 ///
 /// FIXME
-pub fn get_domains() -> Result<(), RelayerError> {
+pub fn get_domains() -> Result<(), SBError> {
     let key = read_keypair_file("../../verification.so").expect("Read to file");
     Ok(())
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct Profile {
+    github: String,
+    solanaAddress: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct VerificationData {
+    name: String,
+    profiles: Vec<Profile>,
+}
+
+/// get_verification
+///
+/// reads the PRs into the verification list and
+/// mints NFTs if they don't exist already
+pub async fn get_verification() -> Result<(), SBError> {
+    let gh = get_connection().await?;
+
+    let mut pulls = match gh
+        .pulls("sandblizzard", "verification")
+        .list()
+        .state(octocrab::params::State::All)
+        .per_page(100)
+        .send()
+        .await
+    {
+        Ok(pulls) => pulls,
+        Err(err) => return Err(SBError::FailedToFetchPulls(err.to_string())),
+    };
+
+    let verification = gh
+        .repos("sandblizzard", "verifcation")
+        .get_content()
+        .path("./data/")
+        .r#ref("main")
+        .send()
+        .await
+        .unwrap();
+
+    // load the verification file
+    let verification_file = File::open("./data/profile.list.json").unwrap();
+    let dd: VerificationData = serde_json::from_reader(verification_file).unwrap();
+
+    // try generate the users using underdog
+    let profiles = dd.profiles;
+    for profile in profiles {
+        // create user
```

Comment:
Is the intention here to use webhooks and the underdog api to generate the link for claiming the nft and releasing escrow? Or is there an authentication system in the underdog api I'm still getting to in the [docs](https://docs.underdogprotocol.com/operation-sandstorm/workshop-2-what-to-build-with-underdog-api)?

Reply:
> Is the intention here to use webhooks and the underdog api to generate the link for claiming the nft and releasing escrow?

We can just mint the NFT without them signing anything. Of course, that means that we have to pay the minting cost. So I was just thinking to have one NFT per Username. So you would have 1 NFT for you github profile that identifies you. This NFT can be used across all bounties you might ever create or solve 

>or is there an authentication system in the underdog api I'm still getting to in the [docs](https://docs.underdogprotocol.com/operation-sandstorm/workshop-2-what-to-build-with-underdog-api)?

the relayer needs to authenticate against the underdog api